### PR TITLE
Fix translucent objects when the texture is RGBA

### LINKED ISSUE #332: --texture-base-color with opacity and depth peeling is not correct
**Describe the bug**
Using a texture base color with opacity and depth peeling, without setting --opacity, cause a wrong rendering

**To Reproduce**
 1. Open the file using `f3d --dry-run WaterBottle.glb --texture-base-color=/home/glow/f3d/src/testing/data/albedo_mod.png --depth-peeling --geometry-only` : 
albedo_mod

 2. wrong rendering:
 
wrong

 4. Using the following provide a correct rendering: `f3d --dry-run WaterBottle.glb --texture-base-color=/home/glow/f3d/src/testing/data/albedo_mod.png --depth-peeling --geometry-only --opacity=0`

**Expected behavior**
Correct translucent rendering, like this:
right

5. The --opacity parameter is not taken into account at all, apart from enabling translucentsy, it should multiply.

### PR DESCRIPTION
**Pull request by Meakk**
Fix #332

### CODE REVIEW

**Comment by mwestphal** [comment]
please rebase on master and add a `TestTextureColor` test.

**Review by mwestphal — CHANGES_REQUESTED** [review]
please rebase on master and add a TestTextureColor test.

**Comment by Meakk** [comment]
Done, please review again @mwestphal

File: cmake/testing.cmake [hunk 1]
```diff
@@ -145,7 +145,7 @@ f3d_test(NAME TestVolumeNonScalars DATA waveletArrays.vti ARGS -vb --scalars=Ran
 f3d_test(NAME TestTextureNormal DATA WaterBottle.glb ARGS --geometry-only --texture-normal=${CMAKE_SOURCE_DIR}/testing/data/normal.png --normal-scale=0.1 DEFAULT_LIGHTS)
 f3d_test(NAME TestTextureMaterial DATA WaterBottle.glb ARGS --geometry-only --texture-material=${CMAKE_SOURCE_DIR}/testing/data/red_mod.jpg --roughness=1 --metallic=1 DEFAULT_LIGHTS)
 f3d_test(NAME TestTextureMaterialWithOptions DATA WaterBottle.glb ARGS --geometry-only --texture-material=${CMAKE_SOURCE_DIR}/testing/data/red_mod.jpg --roughness=0.5 --metallic=0.5 DEFAULT_LIGHTS)
-f3d_test(NAME TestTextureColorWithOptions DATA WaterBottle.glb ARGS --geometry-only --texture-base-color=${CMAKE_SOURCE_DIR}/testing/data/albedo_mod.png --color=1,1,0 --opacity=0 --depth-peeling DEFAULT_LIGHTS) # TODO add a texture-base-color test without --opacity and improve current test once https://github.com/f3d-app/f3d/issues/332 is fixed
+f3d_test(NAME TestTextureColorWithOptions DATA WaterBottle.glb ARGS --geometry-only --texture-base-color=${CMAKE_SOURCE_DIR}/testing/data/albedo_mod.png --color=1,1,0 --depth-peeling DEFAULT_LIGHTS)
```
Comment (mwestphal) [inline cmake/testing.cmake:148]: maybe something like that:
```
 f3d_test(NAME TestTextureColor DATA WaterBottle.glb ARGS --geometry-only --texture-base-color=${CMAKE_SOURCE_DIR}/testing/data/albedo_mod.png --depth-peeling DEFAULT_LIGHTS)

 f3d_test(NAME TestTextureColorWithOptions DATA WaterBottle.glb ARGS --geometry-only --texture-base-color=${CMAKE_SOURCE_DIR}/testing/data/albedo_mod.png --color=1,1,0 --opacity=0.8 --depth-peeling DEFAULT_LIGHTS)
 ```

**Comment by mwestphal** [comment]
So the multiplication issue is not fixed ?

**Comment by Meakk** [comment]
> So the multiplication issue is not fixed ?

Fixed here: https://gitlab.kitware.com/vtk/vtk/-/merge_requests/9467

**Comment by Meakk** [comment]
The test will be fixed when the VTK MR will be merged

File: cmake/testing.cmake [hunk 2]
```diff
@@ -157,6 +157,11 @@ f3d_test(NAME TestAnimationIndex DATA InterpolationTest.glb ARGS --animation-ind
 f3d_test(NAME TestNonExistentFile DATA nonExistentFile.vtp ARGS --filename WILL_FAIL)
 f3d_test(NAME TestUnsupportedFile DATA unsupportedFile.dummy ARGS --filename WILL_FAIL)
 
+# color texture with opacity needs https://gitlab.kitware.com/vtk/vtk/-/merge_requests/9467
+if(VTK_VERSION VERSION_GREATER_EQUAL 9.1.202208XX)
```
Comment (Meakk) [inline cmake/testing.cmake:161]:
```suggestion
if(VTK_VERSION VERSION_GREATER_EQUAL 9.1.20220811)
```

File: cmake/testing.cmake [hunk 3]
```diff
@@ -157,6 +157,11 @@ f3d_test(NAME TestAnimationIndex DATA InterpolationTest.glb ARGS --animation-ind
 f3d_test(NAME TestNonExistentFile DATA nonExistentFile.vtp ARGS --filename WILL_FAIL)
 f3d_test(NAME TestUnsupportedFile DATA unsupportedFile.dummy ARGS --filename WILL_FAIL)
 
+# color texture with opacity needs https://gitlab.kitware.com/vtk/vtk/-/merge_requests/9467
+if(VTK_VERSION VERSION_GREATER_EQUAL 9.1.20220811)
```
Comment (Meakk) [inline cmake/testing.cmake:161]:
```suggestion
if(VTK_VERSION VERSION_GREATER_EQUAL 9.2.20220811)
```

**Review by mwestphal — APPROVED** [review]

**Comment by mwestphal** [comment]
+1

**Pull request merged**